dnf5-plugins/config-manager_plugin/addrepo.cpp

```diff
@@ -116,9 +96,11 @@ std::string generate_repoid_from_url(const std::string & url) {
     // Limits the length of the repository id.
     // Copies the first and last 100 characters. The substring in between is replaced by a crc32 hash.
     if (ret.size() > 250) {
-        std::string_view tmp{ret};
+        unsigned int sz = static_cast <unsigned int> (ret.size());
+        char middle[sz - 200]{};
+        ret.copy(middle, sz - 200, 100);
         ret = fmt::format(
-            "{}-{:08X}-{}", tmp.substr(0, 100), crc32(tmp.substr(100, tmp.size() - 200)), tmp.substr(tmp.size() - 100));
+            "{}-{:08X}-{}", ret.substr(0, 100), crc32(0, (unsigned char *) middle, sz - 200), ret.substr(sz - 100));
```

Comment:
crc32_z() has size_t type of the last argument. If you used that, you wouldn't have to use the unsafe static_cast.

Reply:
Ah yes, I'll change that over to crc32_z().

Reply:
I think that now you can change sz variable type to size_t and remove the static_cast<>.